service_darwin.go

```diff
@@ -184,28 +188,79 @@ func (s *darwinLaunchdService) Uninstall() error {
 }
 
 func (s *darwinLaunchdService) Status() (Status, error) {
-	exitCode, out, err := runWithOutput("launchctl", "list", s.Name)
-	if exitCode == 0 && err != nil {
-		if !strings.Contains(err.Error(), "failed with stderr") {
+	var (
+		// Scan output for a valid PID.
+		re        = regexp.MustCompile(`"PID" = ([0-9]+);`)
+		isRunning = func(out string) bool {
+			matches := re.FindStringSubmatch(out)
+			return len(matches) == 2 // `PID = 1234`
+		}
+		canIgnore = func(err error) bool {
+			return err == nil ||
+				strings.Contains(err.Error(),
+					"Could not find service")
+		}
+	)
+
+	// Get output from `list`.
+	controlArgs := []string{"list", s.Name}
+	_, out, err := runWithOutput(controlCmd, controlArgs...)
+	if !canIgnore(err) {
+		return StatusUnknown, err
+	}
+	if isRunning(out) {
+		return StatusRunning, nil
+	}
+
+	// root only sees "system daemons"
+	// `list` will not see "user daemons" (which we likely are).
+	// If we're impersonating another user (like root via `sudo` or setuid bit),
+	// try again as the user ID that called us.
+	var (
+		impersonating       bool
+		targetID            string
+		uid                 = syscall.Getuid()
+		suidStr, sudoExeced = os.LookupEnv("SUDO_UID")
+		euid                = syscall.Geteuid()
+	)
+	if sudoExeced {
+		suid, err := strconv.Atoi(suidStr)
+		if err != nil {
 			return StatusUnknown, err
 		}
+		impersonating = impersonating || suid != euid
```

Comment:
Boolean logic fault: We used to assign `impersonating` in its declaration above, so it might have been true, but now it's always false.
This section is about to be nuked anyway, but pointing it out regardless.